keps/sig-node/4004-deprecate-kube-proxy-version/README.md

```diff
@@ -0,0 +1,261 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecate the `status.nodeInfo.kubeProxyVersion` field of v1.Node
+
+## Motivation
+
+This field is not accurate, the field is set by kubelet, which does not actually know the kube-proxy version
+
+### Goals
+
+Deprecate the `status.nodeInfo.kubeProxyVersion` field and then do the hard work of finding all the places using it and getting them off in time.
+
+### Non-Goals
+
+## Proposal
+
+- Deprecate the `status.nodeInfo.kubeProxyVersion` field of v1.Node
+- Remove the gcp cloud provider check for this field
+
+### User Stories (Optional)
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### Versioned API Change: NodeStatus v1 core
+
+Mark the kubeProxyVersion field as Deprecated
+
+```
+//Deprecated: Kubelet Version reported by the node.
+KubeletVersion string
+```
+
+### NodeStatus Internal Representation
+
+Mark the kubeProxyVersion field as Deprecated
+
+```
+//Deprecated: Kubelet Version reported by the node.
+KubeletVersion string `json:"kubeletVersion" protobuf:"bytes,7,opt,name=kubeletVersion"`
+```
+
+### Test Plan
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+Every code should have complete unit test coverage
+
+##### Integration tests
+
+
+- N/A
+
+##### e2e tests
+
+- Test whether FeatureGate behaves as expected when it is turned on or off
+- If kubeProxyVersion FeatureGate is enabled:
+  - `status.nodeInfo.kubeProxyVersion` will be empty
+- Else:
+  - `status.nodeInfo.kubeProxyVersion` will be not empty 
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: kubeProxyVersion
+  - Components depending on the feature gate: kubelet
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Using the featuregate is the only way to enable/disable this feature.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature should continue to work just fine.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, these will be introduced in the Alpha phase.
```

Comment:
That would be "Yes", not "No" because you're supposed to include the work being done for Alpha in the answers here.

But I think this should be "No" anyway; "tests for feature enablement/disablement" means "tests where you have a running cluster with the feature disabled and then you enable it", or vice versa. And there's no reason to do that here; we need to make sure no one is looking at the deprecated field ever, so we don't need to test what happens when the value of the field _changes_, since if anyone was actually looking at it all, then we're already failing.